src/nccl_ofi_rdma.c

```diff
@@ -2058,6 +2058,15 @@ static inline int free_write_req(nccl_net_ofi_rdma_req_t *req,
 	assert(req->type == NCCL_OFI_RDMA_WRITE);
 	nccl_net_ofi_rdma_send_comm_t *s_comm =
 		(nccl_net_ofi_rdma_send_comm_t *)req->comm;
+	/* free is going to be called inside of test(), which will
+	   happen in a time when NCCL guarantees no other thread will
+	   be accessing the communicator.  So no mutex protections are
+	   required if we do it here.  Better would be to do this as
+	   soon as we get the CQE for this request, but that would
+	   require atomics or locks, which isn't worth it today.  But
+	   if we ever refactor the locking strategy, we should revisit
+	   this. */
+	(s_comm->num_inflight_writes)--;
```

Comment:
Are you sure this is the right (no pun indended) place for this? `free_write_req` is used by `rma_write_impl`, i.e., the `iwrite` interface we made for Neuron.

Reply:
yeah, you're right.  fixed.